lib/grape/middleware/base.rb

```diff
@@ -21,8 +22,38 @@ def call(env)
       def call!(env)
         @env = env
         before
+
         @app_response = @app.call(@env)
-        after || @app_response
+
+        # Merge Headers from API into @header
```

Comment:
This also duplicates some functionality in `InsideRoute`, maybe this can be a module?